src/Display.java

```diff
@@ -0,0 +1,43 @@
+public class Display {
+
+    public static void showDataAnimal(GenericContainer<? extends Animal> container) {
```

Comment:
this method is written correctly
yet -> instead of container.getAnimal() you were supposed to use the container.getT().getAnimalVariable() here -> the task was supposed to show that if you specify that a GenericContainer takes a ? = sth that extends animal, it will always be able to get to the Animal's variables